### ISSUE TITLE
Better explanation of what line_by_line does

### ISSUE THREAD

**Issue by xdrie** [minimaxir/aitextgen#88]
The description in the existing documentation says that it exists to process a single column CSV treating each line as a single entry.

So I used an input file which was simply a text file with a sentence on each line, which corresponds to the format of a single column csv:
```
some sample text example 1
some sample text example 2
```

However, when I ran `generate` after finetuning (using the Colab notebook)

I got outputs that looked like:
```
<|startoftext|>some sample text example 1<|endoftext|>
```

I was under the impression that those special start and stop tokens were automatically handled. Was I using the library wrong in some way? All the training and the outputs were made in the provided fine-tuning Colab notebook.

**Comment by minimaxir** [comment]
Yes, something changed in either transformers or tokenizers. Will fix for 0.4.0

**Comment by minimaxir** [comment]
Can confirm something is messy with how transformers builds tokenizers which is leading to a discrepancy. This may be less easy than anticipated.

<img width="902" alt="Screen Shot 2021-02-15 at 8 54 00 PM" src="https://user-images.githubusercontent.com/2179708/108019923-16b87480-6fd0-11eb-86e7-b9c66da97bf2.png">

**Comment by minimaxir** [comment]
This is only the case for the fast tokenizers: slow tokenizers work fine. May fall back to that for the release.

**Comment by minimaxir** [comment]
Or, alternatively, filter out the tokens at the PyTorch tensor level before decoding. That may be easier, and have other perks too.

**Comment by channelz** [comment]
Looking for guidance .. Trying to train and generate GPT-2 for short texts via aitextgen using CSV and failing. ( longform content generation works fine) Still getting a long text answer with special tokens embedded.   Looking for function similar to how  model="minimaxir/hacker-news" functions. But trained using aitexgen (not simple-gpt2).  Is this the same issue as above?

**Comment by xdrie** [comment]
@channelz probably, yes.

**Comment by minimaxir** [comment]
No; the issue is different than gpt-2-simple (that uses post-processing, which is what the solution here will require). With gpt-2-simple you need to use the `truncate` params.

**Comment by minimaxir** [comment]
0.4.0 is out, which should have fixed this particular weirdness. Let me know if the issue persists.
